client/SMCE_Client.cpp

```diff
@@ -0,0 +1,339 @@
+    // Main loop, handle the command input                                                                                       
+    bool suspended = false;                                                                                                      
+    while(true){                                                                                                                 
+        std::cout << "$>";                                                                                                       
+        std::string input;                                                                                                       
+        std::getline(std::cin,input);                                                                                            
+        board.tick();                                                                                                            
+        if (input == "-h"){ //help menu                                                                                          
+            print_menu();
+
+        }else if (input == "-p"){ //pause or resume                                                                               
+            if(!suspended){                                                                                                      
+                suspended = board.suspend();                                                                                     
+                if(suspended)                                                                                                    
+                    std::cout << "Board paused" << std::endl;                                                                    
+                else                                                                                                             
+                    std::cout << "Board could not be paused" << std::endl;                                                       
+            }else if(suspended){                                                                                                 
+                board.resume();                                                                                                  
+                suspended = false;                                                                                               
+                std::cout << "Board resumed" << std::endl;                                                                       
+            }
+
+        }else if (input == "-rc"){ //recompile
+            std::cout << "Recompiling.." << std::endl; 
+            t_lock.lock(); // aquire lock before recompiling, so uart_listener is forced to wait                                                                                                                                                                        
+            compile_and_start(sketch,toolchain,board,arduino_root_dir);
```

Comment:
a function like this just makes things opaque imo, why not have a compile function and a start function that you just calls after each other? they dont need to be composed.